packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -867,6 +869,11 @@ const _BaseInput: React.ForwardRefRenderFunction<BladeElementRef, BaseInputProps
   const isReactNative = getPlatformType() === 'react-native';
   const defaultRole = hasPopup ? 'combobox' : undefined;
 
+  const inputGroupProps = useInputGroupContext();
+  const isWithinInputGroup = Object.keys(inputGroupProps).length > 0;
```

Comment:
```suggestion
  const { isInsideInputGroup } = useInputGroupContext();
```

Lets do this instead? because if someone adds default value to input group context in future, this will unknowingly break. 

You can check this-

https://github.com/razorpay/blade/blob/a72944305f0f1374b50a25e8ef7abb52b81fe792/packages/blade/src/components/AnimateInteractions/AnimateInteractionsProvider.tsx#L3-L14